app/helpers/provider/admin/dashboards_helper.rb

```diff
@@ -19,17 +19,17 @@ def sign_class(widget)
     widget.percentual_change > 0 ? 'u-plus' : 'u-minus'
   end
 
-  def dashboard_collection_link(singular_name, collection, path, plural = nil, icon_name = nil)
+  def dashboard_collection_link(singular_name, collection, path, opts)
```

Comment:
You can use keyword arguments

`def dashboard_collection_link(singular_name, collection, path, icon_name: nil, plural: nil)`

https://robots.thoughtbot.com/ruby-2-keyword-arguments

Reply:
Wasn't the point to reduce the method's number of arguments?

Reply:
I don't think that this way is reducing the number of arguments but just hiding that we actually have more arguments 🤔 We would need an object then, but I still don't know how. I would have to look at how this is currently working 😄

Reply:
Creating an object is yet another way to hide arguments tho. I think the opts is a clean way to group optional arguments, reducing the signature of the method. But really, the whole point of this PR is to address @hallelujah 's comment -> https://github.com/3scale/porta/pull/78#discussion_r223078010 So.. I'm confused now.

Reply:
His comment doesn't tell me that the goal is to have less params but that the goal is that the order to call to those params doesn't really matter (which is what you accomplish both with a hash and with what Hery suggested here).

About being the same with an object (so different methods), not really imo, for 2 reasons that I have in mind:
1. Being in a separate method means each method does independently less things.
2. As we have less possible paths for each method, and we have more than 1 method, it is easier to unit test them individually.
So doing it differently means changing the current logic so it is not hiding stuff, it is not the same. But it is clearly not easy to do either 😄

Reply:
You're right I misunderstood Hery's comment, so I'll remove `opts` and use the named arguments.
As for the object discussion, I don't this method is complex enough to justify using an object. I totally agree with you point, however for this particular case, it wouldn't make sense.